src/network/core_utils.rs

```diff
@@ -37,6 +37,38 @@ impl CoreUtils {
         final_slice.join(":")
     }
 
+    fn decode_address_from_hex(input: &str) -> Result<Vec<u8>, std::io::Error> {
+        let mut array = [0u8; 6];
+
+        let mut nth = 0;
+        for byte in input.split(|c| c == ':' || c == '-') {
+            if nth == 6 {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    "invalid mac length".to_string(),
+                ));
+            }
+
+            array[nth] = match u8::from_str_radix(byte, 16) {
+                Ok(value) => value,
+                Err(err) => {
+                    return Err(std::io::Error::new(std::io::ErrorKind::Other, err));
```

Comment:
also add some context to the error here, e.g failed to parse mac address

Reply:
resolved